conan/tools/microsoft/msbuilddeps.py

```diff
@@ -84,6 +88,14 @@ def generate(self):
         # config_filename
         # TODO: This is duplicated in write_generators() function, would need to be moved
         # to generators and called from there
+        exclude_code_analysis = self._conanfile.conf["tools.microsoft.msbuilddeps"].exclude_code_analysis
+        if exclude_code_analysis:
+            if isinstance(exclude_code_analysis, str):
+                self.exclude_code_analysis = exclude_code_analysis.split(";")
+        if self.exclude_code_analysis in [["True"], True]:
+            self.exclude_code_analysis = ["*"]
+        elif self.exclude_code_analysis in [["False"], False]:
+            self.exclude_code_analysis = None
```

Comment:
By definition exclude_code_analysis is always ``None`` or a string, this seems could be greatly simplified, maybe something like this:

```suggestion
        exclude_code_analysis = self._conanfile.conf["tools.microsoft.msbuilddeps"].exclude_code_analysis
        if exclude_code_analysis is not None:
                self.exclude_code_analysis = eval(exclude_code_analysis)
                assert isinstance(self.exclude_code_analysis, (bool, list))
```

The idea is not to reinvent a syntax to create a list of things, just use python syntax and eval()